src/oss/langchain-mcp.mdx

```diff
@@ -356,17 +384,87 @@ app.listen(PORT, () => {
 :::
 
 :::python
-## Additional resources
+## Expose LangChain tools via MCP
 
-* [MCP documentation](https://modelcontextprotocol.io/introduction)
-* [MCP Transport documentation](https://modelcontextprotocol.io/docs/concepts/transports)
-* [langchain_mcp_adapters](https://github.com/langchain-ai/langchain-mcp-adapters)
+You can also expose existing LangChain tools through an MCP server using the `to_fastmcp` function. This allows you to make your LangChain tools available to any MCP client.
+
+```python
+from langchain_core.tools import tool
+from langchain_mcp_adapters.tools import to_fastmcp
+from mcp.server.fastmcp import FastMCP
+
+
+@tool
+def add(a: int, b: int) -> int:
+    """Add two numbers"""
+    return a + b
+
+@tool
+def get_user_info(user_id: str) -> str:
+    """Get information about a user"""
+    return f"User {user_id} is active"
+
+
+# Convert LangChain tools to FastMCP
+fastmcp_tools = [to_fastmcp(add), to_fastmcp(get_user_info)]
+
+# Create MCP server with the converted tools
+mcp = FastMCP("LangChain Tools", tools=fastmcp_tools)
+mcp.run(transport="stdio")
+```
+
+## Stateless architecture
+
+The MCP client is **stateless by default**, which provides several important benefits:
+
+- **Scalability**: Web servers can handle requests without maintaining session state, making horizontal scaling simpler
+- **Reliability**: No risk of losing state when servers restart or crash
+- **Simplicity**: Eliminates the complexity of state synchronization across multiple server instances
+- **Resource efficiency**: No memory overhead from storing session data
+
+Each tool invocation creates a fresh MCP `ClientSession`, executes the tool, and then cleans up. This ensures that:
+- Tools don't interfere with each other
+- Memory usage remains constant
+- There are no state-related bugs between tool calls
+
+### Using session objects for state
+
+If you need to maintain state across tool calls (such as for conversation context or user sessions), use session objects in your application layer:
+
+```python
+from langchain_mcp_adapters.client import MultiServerMCPClient
+from langgraph.prebuilt import create_react_agent
+
+class StatefulAgent:
```

Comment:
This isn't the correct -- it's not the statefulness that MCP refers to

Statefulness is built into the MCP protocol and a session id gets assigned by the server to the client for servers that support statefulness.

---

The session object is a session that's defined by the python MCP SDK. We have lower utility functions that accept the session and can generate stateful tools from it. Take a look at the implementation quikcly?

---
If you want you can remove this entire section, I can add it later. I wasn't imagining more than ~1 sentence + showing the relevant standalone functions